flink-clients/src/main/java/org/apache/flink/client/cli/PythonShellParser.java

```diff
@@ -0,0 +1,288 @@
+			e.printStackTrace();
+			System.exit(1);
+		}
+	}
+
+	/**
+	 * Prints the error message and help for the client.
+	 *
+	 * @param msg error message
+	 */
+	private static void printError(String msg) {
+		System.err.println(msg);
+		System.err.println("Valid cluster type are \"local\", \"remote <hostname> <portnumber>\", \"yarn\".");
+		System.err.println();
+		System.err.println("Specify the help option (-h or --help) to get help on the command.");
+	}
+
+	/**
+	 * Prints the help for the client.
+	 */
+	private static void printHelp() {
+		System.out.print("Flink Scala Shell\n");
+		System.out.print("Usage: start-scala-shell.sh [local|remote|yarn] [options] <args>...\n");
+		System.out.print('\n');
+		printLocalHelp();
+		printRemoteHelp();
+		printYarnHelp();
+		System.out.println("-h | --help");
+		System.out.println("      Prints this usage text");
+	}
+
+	private static void printYarnHelp() {
+		HelpFormatter formatter = new HelpFormatter();
+		formatter.setLeftPadding(5);
+		formatter.setWidth(80);
+		System.out.println("Command: yarn [options]");
+		System.out.println("Starts Flink scala shell connecting to a yarn cluster");
+		formatter.printHelp(" ", yarnOptions);
+	}
+
+	private static void printRemoteHelp() {
+		HelpFormatter formatter = new HelpFormatter();
+		formatter.setLeftPadding(5);
+		formatter.setWidth(80);
+		System.out.println("Command: remote [options] <host> <port>");
+		System.out.println("Starts Flink scala shell connecting to a remote cluster");
+		System.out.println("  <host>");
+		System.out.println("        Remote host name as string");
+		System.out.println("  <port>");
+		System.out.println("        Remote port as integer");
+		System.out.println();
+		formatter.printHelp(" ", remoteOptions);
+	}
+
+	private static void printLocalHelp() {
+		HelpFormatter formatter = new HelpFormatter();
+		formatter.setLeftPadding(5);
+		formatter.setWidth(80);
+		System.out.println("Command: local [options]");
+		System.out.println("Starts Flink scala shell with a local Flink cluster");
+		formatter.printHelp(" ", localOptions);
+	}
+
+	/**
+	 * Constructs yarn options. The python shell option will add prefix 'y' to align yarn options in {@link CliFrontendParser}.
+	 *
+	 * @param options     Options that will be used in `flink run`.
+	 * @param yarnOption  Python shell yarn options.
+	 * @param commandLine Parsed Python shell parser options.
+	 */
+	private static void constructYarnOption(List<String> options, Option yarnOption, CommandLine commandLine) {
+		if (commandLine.hasOption(yarnOption.getOpt())) {
+			options.add("-y" + yarnOption.getOpt());
+			options.add(commandLine.getOptionValue(yarnOption.getOpt()));
+		}
+	}
+
+	/**
+	 * Parses Python shell yarn options and transfer to yarn options which will be used in `flink run` to
+	 * submit flink job.
+	 *
+	 * @param args Python shell yarn options.
+	 * @return Yarn options usrd in `flink run`.
+	 */
+	private static List<String> parseYarn(String[] args) {
```

Comment:
The yarn mode doesn't work: The following exception will be thrown:
JAR file does not exist: yarn-cluster